File: HNRZ Salary Manager/app/Models/Jabatan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

#[Fillable(['name', 'salary'])]
class Jabatan extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = ['name', 'salary', 'description'];

    protected $casts = [
        'salary' => 'integer',
    ];

    public function employees()
    {
        return $this->hasMany(Employee::class, 'jabatan_id');
    }
}

File: hnrz_salary_manager/lib/models/jabatan.dart
class Jabatan {
  final int id;
  final String name;
  final int salary;
  final String description;

  Jabatan({
    required this.id,
    required this.name,
    required this.salary,
    required this.description,
  });

  factory Jabatan.fromJson(Map<String, dynamic> json) {
    return Jabatan(
      id: json['id'],
      name: json['name'],
      salary: json['salary'],
      description: json['description'],
    );
  }
}